wolfcrypt/src/blake2s.c

```diff
@@ -132,24 +136,27 @@ int blake2s_init( blake2s_state *S, const byte outlen )
   store32( &P->node_offset, 0 );
   P->node_depth    = 0;
   P->inner_length  = 0;
-  XMEMSET( P->reserved, 0, sizeof( P->reserved ) );
   XMEMSET( P->salt,     0, sizeof( P->salt ) );
   XMEMSET( P->personal, 0, sizeof( P->personal ) );
 #else
-  XMEMSET( P, 0, sizeof( *P ) );
+  XMEMSET( (blake2s_param *)P, 0, sizeof( *P ) );
```

Comment:
Why is the cast necessary?

Reply:
compiler warning otherwise, around the `volatile` qualifier.